### PR TITLE
Implement `ascending` arg for `sort_values`

### PR DESCRIPTION
**Pull request by charlesbluca**
This, along with multi-column sorting and null positioning, are features of table sorting in dask-sql that could potentially be moved upstream to simplify the sorting logic there.

- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

File: dask/dataframe/shuffle.py [hunk 1]
```diff
@@ -744,18 +760,28 @@ def collect(p, part, meta, barrier_token):
...
 
 
-def set_partitions_pre(s, divisions):
+def set_partitions_pre(s, divisions, ascending=True):
     try:
-        partitions = divisions.searchsorted(s, side="right") - 1
+        if ascending:
+            partitions = divisions.searchsorted(s, side="right") - 1
+        else:
+            partitions = -divisions.searchsorted(s, side="right") % (len(divisions) - 1)
```
Comment (jsignell) [inline dask/dataframe/shuffle.py:768]: It took me a bit to figure out what this is doing. This seems easier to understand and I think it's equivalent in this case:

```suggestion
            partitions = len(divisions) - 1 - divisions.searchsorted(s, side="right")
```

Comment (charlesbluca) [reply dask/dataframe/shuffle.py:768]: Thanks @jsignell! Yeah that expression is a lot simpler and seems to accomplish the same thing

File: dask/dataframe/shuffle.py [hunk 2]
```diff
@@ -744,18 +760,28 @@ def collect(p, part, meta, barrier_token):
...
         partitions[s.isna()] = 0
         not_null = s.notna()
-        partitions[not_null] = divisions.searchsorted(s[not_null], side="right") - 1
-    partitions[(s >= divisions.iloc[-1]).values] = len(divisions) - 2
+        if ascending:
+            partitions[not_null] = divisions.searchsorted(s[not_null], side="right") - 1
+        else:
+            partitions[not_null] = -divisions.searchsorted(
+                s[not_null], side="right"
+            ) % (len(divisions) - 1)
```
Comment (jsignell) [inline dask/dataframe/shuffle.py:781]: Same comment about just using subtraction :)

**Comment by jsignell** [comment]
This looks good to me and the failing test is just #8137

**Comment by jsignell** [comment]
Thanks for this @charlesbluca!

**Pull request merged**